common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesAccountCredentialCache.java

```diff
@@ -103,21 +103,37 @@ public SharedPreferencesAccountCredentialCache(
     }
 
     @Override
-    public synchronized void saveAccount(@NonNull final AccountRecord account) {
+    public synchronized void saveAccount(@NonNull final AccountRecord accountToSave) {
         Logger.verbose(TAG, "Saving Account...");
-        Logger.verbose(TAG, "Account type: [" + account.getClass().getSimpleName() + "]");
-        final String cacheKey = mCacheValueDelegate.generateCacheKey(account);
+        Logger.verbose(TAG, "Account type: [" + accountToSave.getClass().getSimpleName() + "]");
+        final String cacheKey = mCacheValueDelegate.generateCacheKey(accountToSave);
         Logger.verbosePII(TAG, "Generated cache key: [" + cacheKey + "]");
-        final String cacheValue = mCacheValueDelegate.generateCacheValue(account);
+
+        // Perform any necessary field merging on the Account to save...
+        final AccountRecord existingAccount = getAccount(cacheKey);
+
+        if (null != existingAccount) {
+            accountToSave.mergeAdditionalFields(existingAccount);
```

Comment:
Although this is more straightforward to do with Accounts (as opposed to Credentials), I'm thinking I will decline this change so that Accounts & Credentials are merged/written using the same logic -- rationale for credentials provided [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1224#discussion_r573143221). I appreciate the desirability of immutable state, though unless there's a compelling reason, I'm thinking leave as is

Reply:
I'll resolve this issue, feel free to unresolve and/or submit proposed changes if you feel strongly about this

Reply:
Yeah, this is too painful to fix.